modules/http/http_core.c

```diff
@@ -269,15 +271,73 @@ static int http_create_request(request_rec *r)
     return OK;
 }
 
-static void http_pre_read_request(request_rec *r, conn_rec *c)
+static void h1_pre_read_request(request_rec *r, conn_rec *c)
 {
     if (!r->main && !r->prev
         && !strcmp(AP_PROTOCOL_HTTP1, ap_get_protocol(c))) {
+        if (r->proxyreq == PROXYREQ_NONE) {
+            ap_add_input_filter_handle(ap_h1_request_in_filter_handle,
+                                       NULL, r, r->connection);
+        }
         ap_add_output_filter_handle(ap_h1_response_out_filter_handle,
                                     NULL, r, r->connection);
     }
 }
 
+static int h1_post_read_request_early(request_rec *r)
+{
+    if (!r->main && !r->prev && r->proto_num <= HTTP_VERSION(1,1)) {
+        /* HTTP1_BODY_IN takes care of chunked encoding and content-length.
+         */
+        ap_add_input_filter_handle(ap_h1_body_in_filter_handle,
+                                   NULL, r, r->connection);
+    }
+    return OK;
+}
```

Comment:
Install the body filter very early, after a request has been read so that anyone added later can rely on its handling.